### PR TITLE
docs: Describe log analysis for reconcile optimization

### PR DESCRIPTION
**Pull request by andrii-korotkov-verkada**
Documents #18923
Add the info about existing and new logs that are being added for reconciliation.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [x] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: docs/operator-manual/reconcile.md [hunk 1]
```diff
@@ -111,3 +111,9 @@ data:
     # actually changing in content.
     - .status.conditions[].lastTransitionTime
 ```
+
+### Analyzing logs to find slow steps of reconciliation
```
Comment (agaudreault) [inline docs/operator-manual/reconcile.md:115]: I am not sure the comment provide enough information to significantly help debugging. The datadog example depends on the underlying infrastructure of the Argo users and how that infrastructure index logs. Slack is a good place to share splunk/datadog query examples.

What would be interesting at the performance level would be to document all the relevant `_ms` that could help users detect performance problems. For each of them, provide an explanation of what could be the known possible causes of slowness. This doc could most likely lives in a `docs/operator-manual/performance_investigation.md` on it's own.

Comment (christianh814) [reply docs/operator-manual/reconcile.md:115]: +1 on the Arithmetic interpreted note. That would def be useful

**Review by todaywasawesome — CHANGES_REQUESTED** [review]
I agree with @agaudreault's comment though I would still accept this PR and look for additional improvements and eventually a separate section. As is, I think it provides good info to get people started.

File: docs/operator-manual/reconcile.md [hunk 2]
```diff
@@ -111,3 +111,9 @@ data:
...
+
+To query logs in Datadog for example, use `@application:*name` filter and make sure not to have your logs sampled. Also, Datadog processes structured fields much better if ArgoCD outputs them as JSON (configurable via `controller.log.format: json` in `argocd-cmd-params-cm` for application controller). Make sure to include debug logs (configurable via `controller.log.level: debug` for application controller), since a lot of logs with timings are on debug level. Remember, updating `argocd-cmd-params-cm` requires restarting the application controller to take effect.
```
Comment (todaywasawesome) [inline docs/operator-manual/reconcile.md:119]: Discussed with @agaudreault and I think we should just remove this section as it's very specific to Datadog. 

Let's also remove the "You might open an issue and try to fix it or ask the community after finding some inefficiencies."

**Review by todaywasawesome — CHANGES_REQUESTED** [review]

**Pull request closed without merging**